src/main/java/nl/ictm2a4/javagame/screens/GameScreen.java

```diff
@@ -217,4 +220,24 @@ private void registerAchievements() {
         new FirstDoorOpened();
         new LastLevelAchieved();
     }
+
+    public void setCurrentUser(User user){
+        currentUser = user;
```

Comment:
Kan dit ook null zijn?

Reply:
Want als die null kan zijn heb ik persoonlijk liever dat je Optional gebruikt.

Reply:
Ja, dan is hij uitgelogd.

Reply:
Ik zie het voordeel van optionals niet helemaal, leg eens uit.

Reply:
Als return type gebruik je `Optional<User>`
vervolgens doe je in het return `return Optional.ofNullable(user)`

Dan kun je vervolgens getCurrentUser().isPrensent() en #.get() gebruiken.
De eerste om te kijken of er een gebruiker is, de 2e om de user te gebruiken.

Reply:
Heeft dat nog voordelen naast syntactic sugar? Want ik vind het minder duidelijk dan een null check en het gaat wel weer tijd kosten om het aan te passen.

Reply:
Ja zo voorkom je nullPointers. Je IDE geeft het aan als je .get() gebruikt zonder .isPresent() te gebruiken. Dus runtime heeft dit veel voordelen. En ik ben persoonlijk ook niet zo'n fan van vergelijken met null, omdat je het dus wel eens kunt vergeten. En dan gaat het fout.

Reply:
Maar eventueel wil ik dat ook wel voor je aanpassen, maar het is misschien wel goed om mee te oefenen. Dat zul je zelf even moeten beoordelen.

Reply:
Ik zou het voor nu handiger vinden als jij het aanpast

Reply:
Okidoki, ga ik dat doen als je klaar bent met als je pushes.